### PR TITLE
Start separating metadata retrieval into own stream

### PR DESCRIPTION
**Pull request by wbaker-figure**

### CODE REVIEW

**Comment by wbaker-figure** [comment]
closes #10, closes #4

File: lib/src/main/kotlin/io/provenance/eventstream/stream/transformers/StreamBlockImplT.kt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        val blockDatetime = header?.dateTime()
+        val blockResponse = tendermintServiceClient.blockResults(header?.height).result
+        val blockEvents: List<BlockEvent> = blockResponse.blockEvents(blockDatetime)
+        val txEvents: List<TxEvent> = blockResponse.txEvents(blockDatetime) { index: Int -> txHash(index) ?: "" }
+        val streamBlock = StreamBlockImpl(this, blockEvents, txEvents, historical)
+        val matchBlock = matchesBlockEvent(blockEvents, options)
+        val matchTx = matchesTxEvent(txEvents, options)
+
+        // ugly:
+        if ((matchBlock == null && matchTx == null) || (matchBlock == null && matchTx != null && matchTx) || (matchBlock != null && matchBlock && matchTx == null) || (matchBlock != null && matchBlock && matchTx != null && matchTx)) {
```
Comment (jphorec) [inline lib/src/main/kotlin/io/provenance/eventstream/stream/transformers/StreamBlockImplT.kt:49]: I'm confused on this if statement, wouldn't this always return `streamBlock` no matter what?

**Review by jphorec — APPROVED** [review]

File: kafka-coroutine/src/main/kotlin/io/provenance/kafka/coroutine/KafkaSink.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+@OptIn(ExperimentalStdlibApi::class)
+class KafkaSink<K, V>(
+    producerProps: Map<String, Any>,
+    val topicName: String,
+    val kafkaProducer: Producer<K, V> = KafkaProducer(producerProps)
+) {
+
+    suspend fun send(block: V, key: K) {
```
Comment (mtps) [inline kafka-coroutine/src/main/kotlin/io/provenance/kafka/coroutine/KafkaSink.kt:32]: Style: k:v ordering here with params over v:k (backwards)

File: lib/src/main/kotlin/io/provenance/eventstream/stream/transformers/StreamBlockImplT.kt (hunk 1 above)
Comment (mtps) [reply lib/src/main/kotlin/io/provenance/eventstream/stream/transformers/StreamBlockImplT.kt:49]: @jphorec It's pre-existing logic; from my observations, you are correct. This should never return nil unless something fails to fetch.

**Review by mtps — APPROVED** [review]

**Pull request merged**